src/main.rs

```diff
@@ -82,14 +110,113 @@ fn print_error(writer: &mut dyn io::Write, err: &Error) -> Result<(), Error> {
     let mut causes = err.iter_chain();
 
     if let Some(err_msg) = causes.next() {
-        writeln!(writer, "Error: {}", err_msg)?;
+        writeln!(writer, "error: {}", err_msg)?;
     } else {
-        writeln!(writer, "Unknown Error")?;
+        writeln!(writer, "Unknown error")?;
```

Comment:
can causes ever have length 0?

Reply:
> No. But writing it like this instead of unwrapping does not hurt.

I think it does hurt, since it suggests that it might be the case. Leaving out the else case ore putting `unreachable!()` is more meaningful.

Reply:
+1 for `unreachable!()`